core/src/serve_repair.rs

```diff
@@ -512,21 +530,41 @@ impl ServeRepair {
             } else {
                 stats.handle_requests_staked += 1;
             }
-            decoded_reqs.push((request, from_addr, *stake));
+
+            decoded_requests.push(RepairRequestWithMeta {
+                request,
+                from_addr,
+                stake: *stake,
+                whitelisted: false,
```

Comment:
Why not set the `whitelisted` correctly here?

If the goal is to not lock `self.repair_whitelist` while deserializing, I don't think it matters; because we only need a shared lock here, and writes are pretty rare and manual.

Reply:
agreed, it shouldn't matter with a single thread. moved the lock.